fakestorage/upload.go

```diff
@@ -327,10 +327,7 @@ func (s *Server) multipartUpload(bucketName string, r *http.Request) jsonRespons
 func (s *Server) resumableUpload(bucketName string, r *http.Request) jsonResponse {
 	predefinedACL := r.URL.Query().Get("predefinedAcl")
 	contentEncoding := r.URL.Query().Get("contentEncoding")
-	metadata, err := loadMetadata(r.Body)
-	if err != nil {
```

Comment:
Can you add a test that demonstrates this behavior?

Also is it possible to retain this check if we narrow it to ignore specific kinds of errors, e.g., missing metadata?